Error when injecting urls

I am new to Sparkler. I begun with the requirements provided and everything worked well. Then, when I started to inject some urls, I got an issue.

Here are the steps that I followed to inject urls: 

1) cd to the root directory of the project
2) docker run -it sparkler-local
3) /data/solr/bin/solr
4) /data/sparkler/bin/sparkler.sh
5) java -jar sparkler-app-0.1-SNAPSHOT.jar inject -sf seed.txt

The output that it must appear is:
```
2016-06-07 19:22:49 INFO  Injector$:70 [main] - Injecting 2 seeds
>>jobId = sparkler-job-1465352569649

```
But, I got this output: 
`Error: Unable to access jarfile sparkler-app-0.1-SNAPSHOT.jar`

I don't understand what caused this problem and how can I resolve it.  I will be very gratefull  if I get an answer. I look forward to your reply.

Thank you.

Thank you fo your help. The link was very helpful. 
But, I still have some questions since I didn't understand what caused the problem for my case and I didn't find the right outputs yet. 

Do I have to start solr after running this command: docker run -it sparkler-local ? 

When I run (following the link that you sent) : 1) `docker run -it sparkler-local`
                                                                       2) `/data/solr/bin/solr start`
it shows this error:

```
Exception in thread "main" java.nio.file.AccessDeniedException: /data/solr/server/logs/solr_gc.log.0.current -> /data/solr/server/logs/archived/solr_gc.log.0.current
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:457)
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
	at java.nio.file.Files.move(Files.java:1395)
	at org.apache.solr.util.SolrCLI$UtilsTool.archiveGcLogs(SolrCLI.java:3903)
	at org.apache.solr.util.SolrCLI$UtilsTool.runTool(SolrCLI.java:3872)
	at org.apache.solr.util.SolrCLI.main(SolrCLI.java:257)
Failed archiving old GC logs
Exception in thread "main" java.nio.file.AccessDeniedException: /data/solr/server/logs/solr-8983-console.log -> /data/solr/server/logs/archived/solr-8983-console.log
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:457)
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
	at java.nio.file.Files.move(Files.java:1395)
	at org.apache.solr.util.SolrCLI$UtilsTool.archiveConsoleLogs(SolrCLI.java:3932)
	at org.apache.solr.util.SolrCLI$UtilsTool.runTool(SolrCLI.java:3875)
	at org.apache.solr.util.SolrCLI.main(SolrCLI.java:257)
Failed archiving old console logs
Exception in thread "main" java.nio.file.AccessDeniedException: /data/solr/server/logs/solr.log -> /data/solr/server/logs/solr.log.1
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:457)
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
	at java.nio.file.Files.move(Files.java:1395)
	at org.apache.solr.util.SolrCLI$UtilsTool.rotateSolrLogs(SolrCLI.java:3975)
	at org.apache.solr.util.SolrCLI$UtilsTool.runTool(SolrCLI.java:3869)
	at org.apache.solr.util.SolrCLI.main(SolrCLI.java:257)
Failed rotating old solr logs
```

But when I do it in this order:  1) cd to the solr directory
                                               2) `./bin/solr start`
                                               3) `docker run -it sparkler-local`

solr started succesfully. However, when I continue injecting, it still doesn't work since when using this command to inject urls:` java -jar /data/sparkler/sparkler-app/target/sparkler-app-0.1-SNAPSHOT.jar inject -sf seed.txt`

It shows this error:

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at edu.usc.irds.sparkler.Main$.main(Main.scala:47)
	at edu.usc.irds.sparkler.Main.main(Main.scala)
Caused by: java.nio.file.NotDirectoryException: seed.txt is not a directory
	at edu.usc.irds.sparkler.service.Injector.stackListFiles(Injector.scala:126)
	at edu.usc.irds.sparkler.service.Injector.run(Injector.scala:86)
	at edu.usc.irds.sparkler.base.CliTool$class.run(CliTool.scala:34)
	at edu.usc.irds.sparkler.service.Injector.run(Injector.scala:44)
	at edu.usc.irds.sparkler.service.Injector$.main(Injector.scala:147)
	at edu.usc.irds.sparkler.service.Injector.main(Injector.scala)
	... 6 more
```

what I understood from your reply is that you have changed permissions of solr in the project. So, I tried to redownload the project and I run: `mvn clean install`. This was the output:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-war-plugin:2.1.1:war (default-war) on project sparkler-ui: Execution default-war of goal org.apache.maven.plugins:maven-war-plugin:2.1.1:war failed: basedir /home/student/work/sparkler_project/sparkler-master/sparkler-ui/banana/dist does not exist -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

I am having problems to understand the errors and what should I do to resolve that. 
Thank you.

Sorry my fault, git pull and try again please.

Tom

On Mon, Aug 7, 2017 at 3:41 PM, User12300 <notifications@github.com> wrote:

> Thank you fo your help. The link was very helpful.
> But, I still have some questions since I didn't understand what caused the
> problem for my case and I didn't find the right outputs yet.
>
> Do I have to start solr after running this command: docker run -it
> sparkler-local ?
>
> When I run (following the link that you sent) : 1) docker run -it
> sparkler-local
> 2) /data/solr/bin/solr start
> it shows this error:
>
> Exception in thread "main" java.nio.file.AccessDeniedException: /data/solr/server/logs/solr_gc.log.0.current -> /data/solr/server/logs/archived/solr_gc.log.0.current
> 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
> 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
> 	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:457)
> 	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
> 	at java.nio.file.Files.move(Files.java:1395)
> 	at org.apache.solr.util.SolrCLI$UtilsTool.archiveGcLogs(SolrCLI.java:3903)
> 	at org.apache.solr.util.SolrCLI$UtilsTool.runTool(SolrCLI.java:3872)
> 	at org.apache.solr.util.SolrCLI.main(SolrCLI.java:257)
> Failed archiving old GC logs
> Exception in thread "main" java.nio.file.AccessDeniedException: /data/solr/server/logs/solr-8983-console.log -> /data/solr/server/logs/archived/solr-8983-console.log
> 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
> 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
> 	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:457)
> 	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
> 	at java.nio.file.Files.move(Files.java:1395)
> 	at org.apache.solr.util.SolrCLI$UtilsTool.archiveConsoleLogs(SolrCLI.java:3932)
> 	at org.apache.solr.util.SolrCLI$UtilsTool.runTool(SolrCLI.java:3875)
> 	at org.apache.solr.util.SolrCLI.main(SolrCLI.java:257)
> Failed archiving old console logs
> Exception in thread "main" java.nio.file.AccessDeniedException: /data/solr/server/logs/solr.log -> /data/solr/server/logs/solr.log.1
> 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
> 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
> 	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:457)
> 	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
> 	at java.nio.file.Files.move(Files.java:1395)
> 	at org.apache.solr.util.SolrCLI$UtilsTool.rotateSolrLogs(SolrCLI.java:3975)
> 	at org.apache.solr.util.SolrCLI$UtilsTool.runTool(SolrCLI.java:3869)
> 	at org.apache.solr.util.SolrCLI.main(SolrCLI.java:257)
> Failed rotating old solr logs
>
> But when I do it in this order: 1) cd to the solr directory
> 2) ./bin/solr start
> 3) docker run -it sparkler-local
>
> solr started succesfully. However, when I continue injecting, it still
> doesn't work since when using this command to inject urls:java -jar
> /data/sparkler/sparkler-app/target/sparkler-app-0.1-SNAPSHOT.jar inject
> -sf seed.txt
>
> It shows this error:
>
> Exception in thread "main" java.lang.reflect.InvocationTargetException
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at edu.usc.irds.sparkler.Main$.main(Main.scala:47)
> 	at edu.usc.irds.sparkler.Main.main(Main.scala)
> Caused by: java.nio.file.NotDirectoryException: seed.txt is not a directory
> 	at edu.usc.irds.sparkler.service.Injector.stackListFiles(Injector.scala:126)
> 	at edu.usc.irds.sparkler.service.Injector.run(Injector.scala:86)
> 	at edu.usc.irds.sparkler.base.CliTool$class.run(CliTool.scala:34)
> 	at edu.usc.irds.sparkler.service.Injector.run(Injector.scala:44)
> 	at edu.usc.irds.sparkler.service.Injector$.main(Injector.scala:147)
> 	at edu.usc.irds.sparkler.service.Injector.main(Injector.scala)
> 	... 6 more
>
> what I understood from your reply is that you have changed permissions of
> solr in the project. So, I tried to redownload the project and I run: mvn
> clean install. This was the output:
>
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-war-plugin:2.1.1:war (default-war) on project sparkler-ui: Execution default-war of goal org.apache.maven.plugins:maven-war-plugin:2.1.1:war failed: basedir /home/student/work/sparkler_project/sparkler-master/sparkler-ui/banana/dist does not exist -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
>
> I am having problems to understand the errors and what should I do to
> resolve that.
> Thank you.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/USCDataScience/sparkler/issues/124#issuecomment-320682912>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGUeFjm2XyBfyTdJYiPkgQqHzoh603bks5sVyIogaJpZM4OuO08>
> .
>



-- 
Tom Barber
CTO Spicule LTD
tom@spicule.co.uk

http://spicule.co.uk

@spiculeim <http://twitter.com/spiculeim>

Schedule a meeting with me <http://meetme.so/spicule>

GB: +44(0)5603641316
US: +18448141689

<https://leanpub.com/juju-cookbook>

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# ignore everything
+*
+
+# except these
+!bin
+!conf
+!resources
+!sparkler-app/bundles
+!sparkler-app/target/sparkler-app-*.jar
+!sparkler-ui/target/*.war

--- a/.gitignore
+++ b/.gitignore
@@ -52,3 +52,6 @@ sjob-**
 Application Files
 /resources
 felix-cache/
+
+tmp*
+workspace*

--- a/bin/dockler.sh
+++ b/bin/dockler.sh
@@ -28,7 +28,7 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 DIR="$DIR/.."
 
 docker_tag="sparkler-local"
-solr_port=8984
+solr_port=8983
 solr_url="http://localhost:$solr_port/solr"
 spark_ui_port=4041
 spark_ui_url="http://localhost:$spark_ui_port/"
@@ -60,38 +60,23 @@ build_image(){
 ####################
 image_id=`docker images -q "$docker_tag" | head -1`
 if [[ -z "${image_id// }" ]]; then
-    while true; do
-        read -p "Cant find docker image $docker_tag. Do you wish to build docker image? [Y/N]" yn
-        case $yn in
-            [Yy]* ) build_image;
-                    image_id=`docker images -q "$docker_tag" | head -1`
-                    break;;
-            [Nn]* ) exit;;
-            * ) echo "Please answer yes or no.";;
-        esac
-    done
+     echo "Cant find docker image $docker_tag. Going to build it"
+     build_image;
+     image_id=`docker images -q "$docker_tag" | head -1`
 fi
 echo "Found image: $image_id"
 ####################
 
 container_id=`docker ps -q --filter="ancestor=$image_id"`
 if [[ -z "${container_id// }" ]]; then
-    while true; do
-        read -p "No container is running for $image_id. Do you wish to start it? [Y/N]" yn
-        case $yn in
-            [Yy]* ) echo "Staring container"
-                    container_id=`docker run -p "$solr_port":8983 -p "$spark_ui_port:4040" -it --user "$user" -d $image_id`
-                    if [ $? -ne 0 ]; then
-                        echo "Something went wrong :-( Please check error messages from docker."
-                        exit 3
-                     fi
-                    echo "Starting solr server inside the container"
-                    docker exec --user "$user" "$container_id" /data/solr/bin/solr restart -force
-                    break;;
-            [Nn]* ) exit;;
-            * ) echo "Please answer yes or no.";;
-        esac
-    done
+    echo "No container is running for $image_id. Starting it..."
+    container_id=`docker run -p "$solr_port":8983 -p "$spark_ui_port:4040" -it --user "$user" -d $image_id`
+    if [ $? -ne 0 ]; then
+        echo "Something went wrong :-( Please check error messages from docker."
+        exit 3
+     fi
+    echo "Starting solr server inside the container"
+    docker exec --user "$user" "$container_id" /data/solr/bin/solr restart -force
 fi
 ####################
 

--- a/bin/dockler.sh
+++ b/bin/dockler.sh
@@ -32,7 +32,7 @@ solr_port=8984
 solr_url="http://localhost:$solr_port/solr"
 spark_ui_port=4041
 spark_ui_url="http://localhost:$spark_ui_port/"
-user="root"
+user="sparkler"
 
 # check for docker
 command -v docker >/dev/null 2>&1 || { echo "Error: Require 'docker' but it is unavailable." >&2; exit 2; }
@@ -42,9 +42,9 @@ build_image(){
 
     prev_dir="$PWD"
     cd "$DIR"
-    echo "Cleaning workspace to minimize size ..."
-    mvn clean -q
-    echo "Cleaning done..."
+    echo "Building project..."
+    git submodule update --init --recursive
+    mvn package -DskipTests
     cd "$prev_dir"
 
     echo "Building a docker image with tag '$docker_tag' ..."

--- a/bin/dockler.sh
+++ b/bin/dockler.sh
@@ -48,7 +48,7 @@ build_image(){
     cd "$prev_dir"
 
     echo "Building a docker image with tag '$docker_tag' ..."
-    docker build -f "$DIR/sparkler-deployment/docker/Dockerfile" -t "$docker_tag" "$DIR/sparkler-deployment/docker"
+    docker build -f "$DIR/sparkler-deployment/docker/Dockerfile" -t "$docker_tag" "$DIR"
 
     if [ $? -ne 0 ]; then
         echo "Error: Failed"

--- a/conf/sparkler-default.yaml
+++ b/conf/sparkler-default.yaml
@@ -86,8 +86,8 @@ plugins.bundle.directory: ${project.parent.basedir}${file.separator}${project.bu
 # List of activated plugins
 plugins.active:
     - urlfilter.regex
-# - fetcher.jbrowser
-    - fetcher.htmlunit
+#    - fetcher.jbrowser
+#    - fetcher.htmlunit
 
 # All Plugins are listed under this tree
 plugins:

--- a/pom.xml
+++ b/pom.xml
@@ -61,7 +61,7 @@
         <module>sparkler-app</module>
         <module>sparkler-plugins</module>
         <module>sparkler-tests-base</module>
-        <!--<module>sparkler-ui</module>-->
+        <module>sparkler-ui</module>
     </modules>
 
     <dependencyManagement>

--- a/sparkler-deployment/docker/Dockerfile
+++ b/sparkler-deployment/docker/Dockerfile
@@ -17,7 +17,7 @@
 #
 # Pull base image.
 
-FROM ubuntu
+FROM ubuntu:16.04
 
 RUN groupadd --gid 1000 sparkler && \
    useradd -M --uid 1000 --gid 1000 --home /home/sparkler sparkler

--- a/sparkler-deployment/docker/Dockerfile
+++ b/sparkler-deployment/docker/Dockerfile
@@ -23,8 +23,8 @@ RUN groupadd --gid 1000 sparkler && \
    useradd -M --uid 1000 --gid 1000 --home /home/sparkler sparkler
    
    
-RUN apt-get update && \
-    apt-get install -y software-properties-common wget git
+RUN apt-get update &&  apt-get install -y software-properties-common wget lsof emacs-nox
+# RUN apt-get update && apt-get install -y git maven emacs-nox
 
 
 RUN \
@@ -36,44 +36,40 @@ RUN \
   rm -rf /var/lib/apt/lists/* && \
   rm -rf /var/cache/oracle-jdk8-installer
 
+# needed by the Jbrowser
+RUN apt-get update && apt-get install -y --fix-missing openjfx
+
+
 # Define commonly used JAVA_HOME variable
 ENV JAVA_HOME /usr/lib/jvm/java-8-oracle
 
-RUN apt-get update && apt-get install -y git maven emacs-nox
 
 # Define working directory.
 WORKDIR /data
 
-# Copies sparkler
-# ADD ./ /data/sparkler
-RUN git clone https://github.com/USCDataScience/sparkler /data/sparkler
-
-# Setup Solr
+## Setup Solr
 RUN wget http://archive.apache.org/dist/lucene/solr/6.6.0/solr-6.6.0.tgz -O /data/solr.tgz && \
-    cd /data/ && tar xzf /data/solr.tgz 
+    cd /data/ && tar xzf /data/solr.tgz && \
+    mv /data/solr-* /data/solr && rm /data/solr.tgz
 
-RUN mv /data/solr-* /data/solr && rm /data/solr.tgz
-RUN apt-get install lsof
+# add sparkler contents
+ADD ./ /data/sparkler
+
+# Setup bundles - fix the new path
+RUN sed -i 's/plugins.bundle.directory:.*/plugins.bundle.directory: \/data\/sparkler\/sparkler-app\/bundles/g' /data/sparkler/resources/sparkler-default.yaml
+
+
+# create solr core
 RUN /data/solr/bin/solr start -force && \
     /data/solr/bin/solr create_core -force -c crawldb -d /data/sparkler/conf/solr/crawldb/ && \
     /data/solr/bin/solr stop -force
-RUN cd /data/sparkler/sparkler-ui/banana && \
-    git submodule init && \
-    git submodule update && \
-    cd /data/sparkler/sparkler-ui/ && mvn clean package
 
-RUN cd /data/sparkler && mvn clean install -DskipTests
-
-# Setup Sparkler Web Dashboard UI
-RUN cp -r /data/sparkler/sparkler-ui/sparkler-dashboard /data/solr/server/solr-webapp/ && \
+# sparkler  ui with banana dashboard
+RUN cp -r /data/sparkler/sparkler-ui/target/sparkler-ui-*.war /data/solr/server/solr-webapp/sparkler && \
     cp /data/sparkler/conf/solr/sparkler-jetty-context.xml /data/solr/server/contexts/
 
-
-RUN apt-get update && apt-get install -y --fix-missing openjfx  # needed by the Jbrowser
-
-RUN chown -R sparkler:sparkler /data/sparkler
-
-RUN chown -R sparkler:sparkler /data/solr
+# Fix permissions
+RUN chown -R sparkler:sparkler /data/
 
 USER sparkler
 # Define default command.

